code/modules/research/xenoarchaeology/artifact/effects/unknown_effect_cellcharge.dm

```diff
@@ -32,13 +29,16 @@
 
 /datum/artifact_effect/cellcharge/DoEffectPulse()
 	if(holder)
-		for (var/obj/machinery/power/apc/C in range(200, holder))
-			for (var/obj/item/weapon/cell/B in C.contents)
-				B.charge += rand() * 100
-		for (var/obj/machinery/power/battery/S in range (src.effectrange,src))
-			S.charge += 250
-		for (var/mob/living/silicon/robot/M in mob_list)
-			for (var/obj/item/weapon/cell/D in M.contents)
-				D.charge += rand() * 100
-				M << "<span class='notice'>SYSTEM ALERT: Energy boost detected!</span>"
+		for(var/obj/machinery/power/apc/C in range(effectrange, holder))
+			if(C.cell)
+				C.cell.charge += rand() * 100
+		for(var/obj/machinery/power/battery/S in range(effectrange, holder))
+			S.charge += rand() * 100
+		for(var/mob/living/silicon/robot/M in mob_list)
+			if(M.cell)
+				M.cell.charge += rand() * 100
+				if(world.time >= next_message)
+					M << "<span class='notice'>SYSTEM ALERT: Large energy boost detected!</span>"
+		if(world.time >= next_message)
+			next_message = world.time + 100
```

Comment:
This is only 10 seconds, personally think it should be about 30 or more because this is just spam to the average robot